Packs/IRISDFIR/Integrations/IRISDFIR/IRISDFIR.py

```diff
@@ -0,0 +1,138 @@
+        self.headers = {'Authorization': f'Bearer {self.api_key}',
+                        'User-Agent': 'Defined'
+                        }
+
+    def get_last_case_id(self):
+
+        response = requests.get(f'{self.api_endpoint}/manage/cases/list', headers=self.headers,
+                                verify=verify_cert, proxies=proxies
+                                )
+
+        if response.status_code == 200:
+            cases = response.json()
+            if cases:
+                list = []
+                counter = 0
+                for last_case in cases['data']:
+                    list.append(last_case['case_id'])
+                    counter += 1
+
+                return cases['data'][list.index(max(list))]
+            else:
+                return "No cases found."
+        else:
+            return (f"Request failed with status code {response.status_code}.")
+
+    def get_all_cases(self):
+
+        response = requests.get(f'{self.api_endpoint}/manage/cases/list', headers=self.headers,
+                                verify=verify_cert, proxies=proxies
+                                )
+
+        if response.status_code == 200:
+            cases = response.json()
+            if cases:
+                return cases['data']
+            else:
+                return "No cases found."
+        else:
+            return f"Request failed with status code {response.status_code}."
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(dfir_iris):
+    try:
+
+        headers = {'Authorization': f'Bearer {dfir_iris.api_key}', 'User-Agent': 'Defined'}
+
+        response = requests.get(f'{dfir_iris.api_endpoint}/manage/cases/list', headers=headers,
+                                verify=verify_cert, proxies=proxies
+                                )
+
+        if response.status_code == 200:
+            return 'ok'
+        else:
+            if response.status_code == 401:
+                return 'Authorization Error: make sure API Key is correctly set'
+            else:
+                return f'Not able to connect to {dfir_iris.api_endpoint}'
+
+    except DemistoException as e:
+        if 'Forbidden' in str(e):
+            return 'Authorization Error: make sure API Key is correctly set'
+        else:
+            raise e
+
+
+def process_incidents(results_str, args: Dict[str, Any]) -> CommandResults:
+
+    return CommandResults(
+        outputs_prefix='IRIS',
+        outputs_key_field='',
+        outputs=results_str,
+    )
+
+
+''' MAIN FUNCTION '''
+
+
+def main():
+    """ COMMANDS MANAGER / SWITCH PANEL """
+    params = demisto.params()
+    command = demisto.command()
+
+    demisto.info(f'Command being called is {command}')
+    try:
+        # initialized Authentication client
+        api_key = params.get('api_key', {}).get('password', '')
+        api_endpoint = params.get('host')
+        #  ignore proxy
+        os.environ['NO_PROXY'] = api_endpoint.split("https://")[1]
```

Comment:
What is the use case of setting 'NO_PROXY' as an environment variable?
Using split and then accessing an element using an index can cause errors, since the split() function can return an empty list (if the user for instance entered `http` instead of `https`). Use `urllib3.util.parse_url(api_endpoint).host` instead.

Reply:
no proxy removed.